src/tools/base/BaseSegmentationTool.js

```diff
@@ -0,0 +1,82 @@
+import BaseTool from './BaseTool.js';
+import { setToolCursor } from '../../store/setToolCursor.js';
+import external from '../../externalModules.js';
+import { getCursor } from './../../util/segmentation';
+
+class BaseSegmentationTool extends BaseTool {
+  constructor(props, defaultProps = {}) {
+    defaultProps.mixins.push('segmentationAPI');
```

Comment:
Important NOTE: this .push is needed on Base Classes to keep mixins coming from it's children